microservices/front-end/src/components/QuestionFilter.tsx

```diff
@@ -0,0 +1,164 @@
+          <InputLabel id="attempted-label">Status</InputLabel>
+          <Select
+            labelId="attempted-label"
+            id="attempted-select"
+            value={attempted}
+            label="Attempted"
+            onChange={handleAttemptFilterChange}
+          >
+            <MenuItem value="All">All</MenuItem>
+            <MenuItem value="Attempted">Attempted</MenuItem>
+            <MenuItem value="Unattempted">Unattempted</MenuItem>
+          </Select>
+        </FormControl>
+      </Box>
+
+      <Box sx={{ minWidth: 250 }}>
+        <FormControl fullWidth>
+          <InputLabel id="difficulty-label">Level</InputLabel>
+          <Select
+            labelId="difficulty-label"
+            id="difficulty-select"
+            value={difficulty}
+            label="Difficulty"
+            onChange={handleDifficultyFilterChange}
+          >
+            <MenuItem value="All">All Levels</MenuItem>
+            <MenuItem value="Easy">Easy</MenuItem>
+            <MenuItem value="Medium">Medium</MenuItem>
+            <MenuItem value="Hard">Hard</MenuItem>
+          </Select>
+        </FormControl>
+      </Box>
+
+      <Box sx={{ minWidth: 250 }}>
+        <FormControl fullWidth>
+          <InputLabel id="sort-label">Sort By</InputLabel>
+          <Select
+            labelId="sort-label"
+            id="sort-select"
+            value={sortBy}
+            label="Sort By"
+            onChange={handleSortChange} // New handler for sorting
+          >
+            <MenuItem value="">None</MenuItem>
+            <MenuItem value="Title">Title</MenuItem>
+            <MenuItem value="Complexity">Complexity</MenuItem>
+          </Select>
+        </FormControl>
+      </Box>
+
+      {/* Added Search Bar */}
+      <Paper
+        component="form"
+        sx={{
+          p: "2px 4px",
+          display: "flex",
+          alignItems: "center",
+          width: 300,
+          boxShadow: "none",
+          border: "1px solid rgba(0, 0, 0, 0.23)",
+        }}
+      >
+        <InputBase
+          sx={{ ml: 1, flex: 1 }}
+          placeholder="Search Questions"
+          inputProps={{ "aria-label": "search questions" }}
+          value={localSearchQuery}
+          onChange={handleSearchInputChange}
+        />
+        <IconButton
+          type="button"
+          sx={{ p: "10px" }}
+          aria-label="search"
+          onClick={handleSearch}
+        >
+          <SearchIcon />
+        </IconButton>
+      </Paper>
+
+      {/* Display Chip if search query is not empty */}
+      {searchQuery && (
+        <Box sx={{ display: "flex", alignItems: "flex-end" }}>
+          <Chip
+            label={searchQuery}
+            onDelete={() => {
+              setLocalSearchQuery(""); // Clear the search query
+              onSearch(""); // Notify parent component to clear the filter
+            }}
+            variant="outlined"
+            sx={{ ml: 2 }} // Margin for aesthetics
+          />
```

Comment:
I don't think this is what `Chip` was meant for

Yes, LeetCode uses `Chip`s, but theirs works because it shows the user the different filters that are currently applied (e.g., filter by what difficulty, filter by what category, filter by what search term), whereas we use drop down menus to tell the user what attempted status / what complexity the questions are filtered by (i.e., our `Chip` component looks a bit redundant)

---

How LeetCode uses `Chip` (source: https://leetcode.com/problemset/all/?topicSlugs=array&page=1&listId=wpwgkgt&search=merge):

![image](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g37/assets/76687125/b56163a9-7d82-454d-8f37-e21e4453b320)

---

How we are using `Chip`:

![image](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g37/assets/76687125/03373ed4-743e-4ab3-8e24-0a6578a219b4)

---

If you want an `X` button to easily clear the search query, you can put the search icon on the left, the text box in the centre and the `X` button on the right (e.g., similar to YouTube):

![image](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g37/assets/76687125/a4f99cd3-01bc-40da-9f4f-51c8d9b8e844)